cmd/tsbs_generate_queries/databases/mongo/devops-naive.go

```diff
@@ -230,30 +229,47 @@ func (d *NaiveDevops) HighCPUForHosts(qi query.Query, nHosts int) {
 // SELECT DISTINCT ON (hostname) * FROM cpu
 // ORDER BY hostname, time DESC
 func (d *NaiveDevops) LastPointPerHost(qi query.Query) {
-	metrics := devops.GetAllCPUMetrics()
-
 	pipelineQuery := []bson.M{
 		{"$match": bson.M{"measurement": "cpu"}},
-		{"$sort": bson.M{"time": -1}},
 		{
 			"$group": bson.M{
-				"_id":       "$tags.hostname",
-				"last_time": bson.M{"$first": "$time"},
+				"_id":       bson.M{"hostname": "$tags.hostname"},
+				"last_time": bson.M{"$max": "$time"},
+			},
+		},
+		{
+			"$lookup": bson.M{
+				"from": "point_data",
+				"let":  bson.M{"time": "$last_time", "hostname": "$_id.hostname"},
+				"pipeline": []bson.M{
+					{
+						"$match": bson.M{
+							"$expr": bson.M{
+								"$and": []bson.M{
+									{"$eq": []interface{}{"$time", "$$time"}},
+									{"$eq": []interface{}{"$tags.hostname", "$$hostname"}},
+									{"$eq": []interface{}{"$measurement", "cpu"}},
+								},
+							},
+						},
+					},
+					{
+						"$project": bson.M{
+							"time": 0,
+							"tags": 0,
+							"_id":  0,
+						},
+					},
+				},
+				"as": "metrics",
 			},
 		},
-	}
-
-	groupMap := pipelineQuery[2]["$group"].(bson.M)
-	for _, metric := range metrics {
-		groupMap[metric] = bson.M{"$first": "$" + metric}
 	}
 
 	humanLabel := "Mongo last row per host"
 	q := qi.(*query.Mongo)
 	q.HumanLabel = []byte(humanLabel)
 	q.BsonDoc = pipelineQuery
-	q.Opts = &options.AggregateOptions{}
-	q.Opts.SetAllowDiskUse(true)
 	q.CollectionName = []byte("point_data")
```

Comment:
it's a bit jank that the collection is a parameter here but it's also hardcoded into the query text, maybe it doesn't matter but if we are thinking of publishing the benchmarks in this form, it might be worth changing pipelineQuery into a function that takes the collection name

Reply:
Hmm yeah agree that it's not the best style. It's worth noting that the original lastpoint query for the recommended schema (before any of our changes) also had that, so I'm going to go ahead and say that Timescale won't mind this.